### DISCUSSION TITLE
Parent span timestamp is pined at service start moment or the child request sent moment?

### DISCUSSION THREAD

**Discussion by bigbanglwb** [apache/skywalking#9525] [Q&A]
Skywalking agent can avoid modify code so I used it. However, I am confused about when the timestamp is pined? 
Take an example, now I have 2 service, named as "Parent Service" and "Child Service". 

- When a client request come to "Parent Service", it map to a controller ( like Spring boot) , it will do some work and send an request to "Child Service". 
- "Child Service" got the request and return the response to "Parent Service".
- "Parent Service" wait the "Child Service" response and do some work, then it return response to client.

Exit span belongs to "Parent Service", and entry span belongs to "Child Service", so I want to know exit span' s start timestamp is pined at the moment of "parent service" got the client request or the moment of "Parent Service " sent the request to "Child Service". Also, is the same true of the end time?

I have read serval blog and discussion but still confused. I would be very grateful if you could reply my question.
![image](https://user-images.githubusercontent.com/43382319/186625783-45173f2f-a87c-44ab-bcec-66e4d15d9f8c.png)

**Comment by wu-sheng** [comment]
The time is just on the minute level, so, I would not be surprised if they are the same. Even visualizing in `ms`, they could be the same too. 

I am not sure what is your question actually, this has nothing related to skywalking, it is just the reality of the time.

**Reply by bigbanglwb** [reply]
The picture is not my doubt, it just take example for the relationship of "Parent Service" and "Child Service"

**Reply by wu-sheng** [reply]
Then, what is the actual question?

**Reply by bigbanglwb** [reply]
`Sevice A` send an request to`Service B` .  My question is when the` service A` 's exit span is created and ended ?

**Reply by wu-sheng** [reply]
That depends on which plugin you are using. Exit span is created by the HTTP client plugin.

**Comment by pg-yang** [comment]
> Exit span belongs to "Parent Service", and entry span belongs to "Child Service", so I want to know exit span' s start timestamp is pined at the moment of "parent service" got the client request or the moment of "Parent Service " sent the request to "Child Service". Also, is the same true of the end time?

`Parent Service`  include two component usually , receiver  or  sender , like `spring mvc controller` and `apache HttpClient` .
For `Parent Service`, It create a Entry span when receive a request , and create Exist span when send a request by HttpClient

**Reply by bigbanglwb** [reply]
Thank you, your answer is completely solve my question! So can I think that `parent service` exit span will very similar to entry span? Because their send time and recv time is similar. 

I mean, is the `parent service` exit span end timestamp pined when received the `child service` response?

![image](https://user-images.githubusercontent.com/43382319/186632223-3c91ea15-1dd6-487e-922f-4f369714f0ed.png)
As the figure shows, I am confused about why the exit span is much longer than entry span. It just caused by wrong code?

**Reply by pg-yang** [reply]
```
+-------------+         +------------------+            +-------------------------+
|   Browser   |         |  Case Servlet    |            | ContextPropagateServlet |
|             |         |                  |            |                         |
+-----|-------+         +---------|--------+            +------------|------------+
      |                           |                                  |
      |                           |                                  |
      |       WebHttp            +-+                                 |
      +------------------------> |-|         HttpClient             +-+
      |                          |--------------------------------> |-|
      |                          |-|                                |-|
      |                          |-|                                |-|
      |                          |-| <--------------------------------|
      |                          |-|                                +-+
      | <--------------------------|                                 |
      |                          +-+                                 |
      |                           |                                  |
      |                           |                                  |
      |                           |                                  |
      |                           |                                  |
      +                           +                                  +
```

`Case Servlet  ` receive a request will create entrySpan , And close the span when it finish all work .    HttpClient's exitSpan is A `Child span` of entrySpan .   
Generally speaking， HttpClient exist span close when it receive response. 
 And `Case Servlet` entrySpan  Close when it reponse client complete 。
https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/plugin-test/
